binner/binner_main.py

```diff
@@ -75,26 +75,26 @@ def get_smallest(self):
   algorithm
   """
   def show(self):
-    from . import log
-    if self.args.algorithm == "smallest":
-      smallest = self.get_smallest()
-      if smallest:
-           result =dict(smallest=self.get_smallest().to_dict())
+      if self.args.algorithm == "smallest":
+          return self.get_result('smallest')
       else:
-	   result = dict(smallest=False)
-    else:
-      lost_items = []
-      for k, item in self.items.items.iteritems():
-	  if not item.used:
-	      lost_items.append( item.to_dict() )
-      log.debug("Result for PACKED items")
-      result = dict(lost=lost_items,
-		run=dict(id=self.id),
-              packed=self.get_packed_bins())
-
-      log.debug( result )
-	
-
-    return result
-
+          return self.get_result()
 
+  def get_result(self, algorithm=None):
+      """Return processed data in dict form."""
+      if algorithm == "smallest":
+          smallest = self.get_smallest()
+          if smallest:
+              result = dict(smallest=self.get_smallest().to_dict())
+          else:
+              result = dict(smallest=False)
+      else:
+          lost_items = []
+          for k, item in self.items.items.items():
+              if not item.used:
+                  lost_items.append(item.to_dict())
+              result = dict(
```

Comment:
I think this is the reason why the results are not being returned correctly. This line of code should be outside the loop.

Reply:
Corrected the indentation but still fails for item 2

```
{"lost": [{"w": 5, "h": 5, "d": 5, "id": 202}, {"w": 5, "h": 5, "d": 5, "id": 203}, {"w": 5, "h": 5, "d": 5, "id": 204}], "run": {"id": "dcae205c-1c07-403c-b827-ffc5a5a43dc1"}, "packed": [{"bin": {"id": 1, "w": 10, "h": 10, "d": 10, "initial": {"w": 10, "h": 10, "d": 10}}, "slots": [{"id": "65dfb9cf-9211-4372-a539-c1d04510e30e", "item": {"w": 2, "h": 2, "d": 2, "id": 202}, "min_x": 1, "max_x": 3, "min_y": 0, "max_y": 3, "min_z": 0, "max_z": 2}, {"id": "7f00cfd2-ec4c-4944-8385-c9a808ea8cc2", "item": {"w": 2, "h": 2, "d": 2, "id": 203}, "min_x": 0, "max_x": 2, "min_y": 3, "max_y": 2, "min_z": 5, "max_z": 7}, {"id": "c22d8ba0-678f-464a-8e13-36b014616c23", "item": {"w": 2, "h": 2, "d": 2, "id": 204}, "min_x": 0, "max_x": 2, "min_y": 3, "max_y": 2, "min_z": 5, "max_z": 7}]}]}
```